### PR TITLE
Add Rosetta support to multi-module

### LINKED ISSUE #1516: Utilize official master branches in rosetta pulls
**Problem**
In its development the rosetta branch utilized some feature branch code in the `hedera-sdk-go` and `Hedera-mirror-node` repos.

**Solution**
- ~Replace LimeChain’s forked Hedera SDK with the official one. (Ongoing PR for that https://github.com/hashgraph/hedera-sdk-go/pull/87). Once PR is merged and released, remove replace github.com/hashgraph/hedera-sdk-go v0.9.1 => github.com/limechain/hedera-sdk-go v0.9.2-0.20200825132925-ccbc4019e257 in rosetta’s go.mod~ Upgrade to go sdk v2, captured in [1521](https://github.com/hashgraph/hedera-mirror-node/issues/1521)
- Replace LimeChain’s forked hedera-mirror-node repository with the original mirror-node-repo in the “all-in-one” Dockerfile (rosetta/build/Dockerfile) cloner step (RUN git clone https://github.com/LimeChain/hedera-mirror-node.git)

**Alternatives**

**Additional Context**

### PR DESCRIPTION
**Pull request by Nana-EC**
**Detailed description**:
In order to work with Rosetta, Hedera needs to implement the [Rosetta Blockchain Standard](https://github.com/coinbase/rosetta-specifications).
Work has been done to create a working api in the `rosetta` branch. 
This should be merged into master to make the api available in future releases

- Merge `rosetta` branch into master - Contains PR's #984 , #1127 , #1128 , #1129, #1136, #1155, #1149, #1169, #1195, #1233
- Add `pom.xml` to rosetta module. Include `docker-maven-plugin` to support image deployment during CI process
- Add rosetta module to main `pom.xml` 
- Updated `docker-compose.yml` with rosetta service

**Which issue(s) this PR fixes**:
Partially addresses #810 
Fixes #1516 

**Special notes for your reviewer**:
This forms the basis. Additional tasks once this is merge in will include
- Utilizing `github.com/hashgraph/hedera-sdk-go` instead of `github.com/limechain/hedera-sdk-go`
- Utilizing `hedera-mirror-node` instead of LimeChain’s forked `hedera-mirror-node` in "all in one" DOckerfile
- Updating db calls with any post V1.31 db schema changes
- Update to `operations.md` with more documentation
- Adding a helm chart
- Add go to license regex
- Update copy rights with go file coverage
- Add go code to github action code scan
- Add go code to sonar

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: hedera-mirror-rosetta/pom.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    </properties>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.fabric8</groupId>
+                <artifactId>docker-maven-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
```
Comment (steven-sheehy) [inline hedera-mirror-rosetta/pom.xml:29]: Should add a task to integrate go into maven lifecycle.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rosetta/build/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+RUN go build -o rosetta-executable ./cmd
+
+# ---------------------------- Importer/GRPC ----------------------------- #
+FROM openjdk:11.0 as java-builder
+
+RUN apt-get update
+COPY --from=cloner /hedera-mirror-node ./hedera-mirror-node
+RUN cd hedera-mirror-node && ./mvnw clean package -DskipTests -pl '!hedera-mirror-rest, !hedera-mirror-rosetta'
+
+# --------------------------------- REST --------------------------------- #
```
Comment (steven-sheehy) [inline hedera-mirror-rosetta/build/Dockerfile:26]: Please remove. REST API module does not run as part of Rosetta.

File: hedera-mirror-rosetta/build/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+COPY --from=cloner /hedera-mirror-node ./hedera-mirror-node
+WORKDIR /tmp/hedera-mirror-node/hedera-mirror-rosetta
+RUN go build -o rosetta-executable ./cmd
+
+# ---------------------------- Importer/GRPC ----------------------------- #
+FROM openjdk:11.0 as java-builder
+
+RUN apt-get update
+COPY --from=cloner /hedera-mirror-node ./hedera-mirror-node
+RUN cd hedera-mirror-node && ./mvnw clean package -DskipTests -pl '!hedera-mirror-rest, !hedera-mirror-rosetta'
```
Comment (steven-sheehy) [inline hedera-mirror-rosetta/build/Dockerfile:24]: This shouldn't build grpc, monitor, etc. Should just `-pl hedera-mirror-importer`.

File: hedera-mirror-rosetta/pom.xml (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-rosetta/pom.xml:29]: Yes, I called this out in #810 . Will create a separate task.
Was hoping to get it in with this PR but didn't want it to become too large.
[mvn-golang](https://github.com/raydac/mvn-golang) seems like a good starting option

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rosetta/pom.xml (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-rosetta/pom.xml:29]: Created #1523 to capture it

File: hedera-mirror-importer/src/main/resources/db/scripts/init_v1.sql [hunk 4]
```diff
@@ -1,22 +1,49 @@
...
-create user :grpc_user with login password :'grpc_password';
+\set rosetta_user 'mirror_rosetta'
+\set rosetta_password 'mirror_rosetta_pass'
+
+create function init_user(user_name text, user_pass text) returns void as $$
+begin
+    if not exists (select from pg_catalog.pg_roles where rolname = user_name) then
+        execute format(
+                'create user %I with
+                    createrole
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-importer/src/main/resources/db/scripts/init_v1.sql:19]: Updated. Looking forward to v2 where we've sorted this out.

File: pom.xml [hunk 5]
```diff
@@ -38,6 +38,7 @@
         <module>hedera-mirror-monitor</module>
         <module>hedera-mirror-protobuf</module>
         <module>hedera-mirror-rest</module>
+        <module>hedera-mirror-rosetta</module>
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply pom.xml:41]: Added

File: hedera-mirror-rosetta/build/Dockerfile (hunk 2 above)
Comment (Nana-EC) [reply hedera-mirror-rosetta/build/Dockerfile]: Removed

File: hedera-mirror-rosetta/build/Dockerfile (hunk 3 above)
Comment (Nana-EC) [reply hedera-mirror-rosetta/build/Dockerfile]: Updated

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**